src/fullfile.c

```diff
@@ -109,6 +109,33 @@ static double fullfile_query_total_download_size(struct list *files)
 	return total_size;
 }
 
+static int get_cached_fullfile(struct file *file) {
```

Comment:
@otaviobp Since in some cases I am copying content to the statedir, I chose `get_cached_fullfile` instead of `fullfile_exists_or_cached`.